pyintegration/deephaven2/learn/gather.py

```diff
@@ -0,0 +1,90 @@
+# 
+# Copyright (c) 2016 - 2022 Deephaven Data Labs and Patent Pending
+# 
+""" Utilities for gathering Deephaven table data into Python objects """
+import enum
+from typing import Any, Type
+
+import jpy
+import numpy as np
+
+from deephaven2 import DHError
+
+_JGatherer = jpy.get_type("io.deephaven.integrations.learn.gather.NumPy")
+
+
+class MemoryLayout(enum.Enum):
+    """ Memory layouts for an array. """
+
+    ROW_MAJOR = True
+    """ Row-major memory layout."""
+    COLUMN_MAJOR = False
+    """ Column-major memory layout."""
+    C = True
+    """ Memory layout consistent with C arrays (row-major)."""
+    FORTRAN = False
+    """ Memory layout consistent with Fortran arrays (column-major)."""
+
+    def __init__(self, is_row_major):
+        self.is_row_major = is_row_major
+
+
+def _convert_to_numpy_dtype(np_type: Type) -> Type:
+    """ Converts an input type to the corresponding NumPy data type. """
+    if np_type.__module__ == np.__name__:
+        return np_type
+    elif np_type == bool:
+        np_type = np.bool_
+    elif np_type == float:
+        np_type = np.double
+    elif np_type == int:
+        np_type = np.intc
+    else:
+        raise ValueError(f"{np_type} is not a data type that can be converted to a NumPy dtype.")
+    return np_type
+
+
+def table_to_numpy_2d(row_set, col_set, order: MemoryLayout = MemoryLayout.ROW_MAJOR, np_type=np.intc):
```

Comment:
1) return type
2) np_type type
3) Ideally we should annotate row_set and col_set, but I think they are java types without python equivalents... so I think we can't do anything for them.

Reply:
@jjbrosnan Do you think there should be a default for `np_type`?  I might lean to no.

Reply:
There should not be a default type for `np_type`.  I'm not really sure why I set the default to a 32-bit integer in the first place.